### PR TITLE
ngintegration enable codecept

### PR DESCRIPTION
**Pull request by sreekanthpuligadda**
commit 753f190dc4ddfaa59609efdce8f9c63581df5fed
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Fri Aug 11 16:06:36 2023 +0100

    enable ngintegration run only in preview and nightly

commit a837f50f93ac11a4f6dfb0ef28892bbc63df38ba
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 10 13:40:07 2023 +0100

    disabled a11y

commit 4041558b518b2edab5b95649daf62e6445b626dd
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 10 13:12:19 2023 +0100

    enabled a11y tests

commit f7ef4d70238a36c09d7f04b9426e5d71ecedb15c
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 10 12:20:05 2023 +0100

    mock response capture excluded in e2e test runs

commit 05ae270222f151b3ae4e6831a9eaaa82dd6c5765
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 10 11:14:22 2023 +0100

    codecept user auth token extract fix

commit 85e12d0ef6956eeaf68f1a3a189a5434372ecdef
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 9 16:17:06 2023 +0100

    create hearing base feature file setup

commit 274f06e6c50268e84a9e8aafc88c8edc50b857f8
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 9 14:49:25 2023 +0100

    disabled a11y for config issue

commit 01de8b3503d676923a74b0e01d1c037456d81879
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 9 13:07:47 2023 +0100

    hearing Tab mock data setup and sample scenario done

commit afb3a284faa683f0aa259a51d61493c4004d7dc3
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 9 09:53:08 2023 +0100

    a11y config fix

commit c3d264c277ddd898bf4a9b26741ba839e9dc5bac
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Aug 8 17:19:20 2023 +0100

    codecept conf fix

commit 4e57d1cf1c2aee9fc78ff6fbc3ab8441cd173121
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Aug 8 13:50:33 2023 +0100

    codecept config fix

commit 9e461fb22d92b6f1fbec4a48523b5b93a2830898
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Aug 8 13:21:41 2023 +0100

    functional config update

commit ad9bcd3d32010b7b74fcc2a69446002b3b0d6df7
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Mon Aug 7 15:52:07 2023 +0100

    ngIntegration test fixes

commit 8561f8b1d44c1be059cce3fbe5604610c5487e53
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Mon Aug 7 10:42:06 2023 +0100

    ngIntegration create hearing mock data setup

commit ab357641e88713841a739ce045f332958e46602a
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Fri Aug 4 14:32:12 2023 +0100

    hearing mock setup in backendMock

commit 6e50b1245bc5532a75bba6e83d46483348deef02
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Fri Aug 4 09:50:07 2023 +0100

    test updates

commit 080eae91bf5f71e0bcb0b0c3285f5f2b9c9f419f
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 3 17:51:25 2023 +0100

    test run analysis report

commit 8ffc2379968d363d923d874bf9193aba4cd0ae58
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 3 13:55:16 2023 +0100

    a11y enabled

commit 2b8c1f0b9cf954c323b9fe3bd8b63db8016c67ea
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Aug 3 10:53:20 2023 +0100

    ngintegration test fix

commit 131ec3ce39ba2ae3bf0a47b4c8bc0c735b8d9aec
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 2 13:30:25 2023 +0100

    CVE fix

commit fbb30f1e47d699fad875cebb0d4b4cff78c0daf3
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 2 13:00:02 2023 +0100

    temp disable pa11y tests

commit 80ad9906e44bb3d4d214810172781e057f9588fa
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Aug 2 12:33:11 2023 +0100

    bgIntegration booking UI test fixes

commit f42e5b149a21e6d1054734d5111a4a03b69dec12
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Aug 1 14:52:01 2023 +0100

    test fixes applied

commit d6a49969d1492775449d143c7d99968d0d12b12b
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Mon Jul 31 09:34:07 2023 +0100

    test updates

commit 0cc2869fe7777cfb6809b42de7a59432c351cb33
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Fri Jul 28 14:18:41 2023 +0100

    test updates

commit 008f0720ddaba90e724dfe526985b7fcbe2aa44d
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Fri Jul 28 10:50:27 2023 +0100

    ngIntegration heacingCase type setup added

commit 5acb64ee4ad23d0785c0acfdfe9e217582b087c1
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Jul 27 11:13:55 2023 +0100

    added ccd case api mocks

commit 49a7a04ede696a3025840aed887bb96d1eb7804c
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Jul 25 10:37:04 2023 +0100

    added feature logs to separate files

commit 4c4e953b5a380b2a871ded50f4ac64350857367f
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Tue Jul 25 10:30:13 2023 +0100

    ngintegration test fix

commit b2c9e7bd208e9891dfca686f84afc227e6577dd1
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Jul 20 11:57:20 2023 +0100

    codecept conf test type condition check added

commit 9ce54794d58b66429a41ed23bab8415df117ef23
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Thu Jul 20 11:10:13 2023 +0100

    ngintegration framework updates

commit e1e173e70c526bf834ba8fe5547e5a4b1b6c2fe7
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Jul 12 20:56:40 2023 +0100

    feature file syntax type error fix

commit b5cb5b6269752b4709013eb9909107c9ada97345
Author: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>
Date:   Wed Jul 12 13:03:23 2023 +0100

    mock user role category fix

commit 9f472cf1808849a76f4a8b98865d2e3312160175
Author: sreekanthpuligadda <52752138+sreekanthpuligadda@users.noreply.github.com>
Date:   Tue Jul 11 23:05:00 2023 +0100

    ngintegration codecept enabled

    * codecept_migration on latest master

    * smoke test command with codecept fixed

    * test updates

    * a11y test enabled

    * a11y test report fixed

    * staff UI test fix

    * test updates

    * disabled ngIntegration tomimprovmenet stability

    * global search test update

    * enable xuiIntegration tests

    * Test updated on Priority column display to jufge after R4

    * ng Integration test update

    * ngIntegration test updates

    * ngIntegration test fixes

    * test fix

    * functional parallel execution thread count increased

    * type error fix

    * Added test scenario for invalid date error msg (#3006)

    * caseDetailsCheck now validates all tabs (#3040)

    * test to verify search filters are retained (#3042)

    * CVE suppress

    * Media viewer functionality (#3009)

    * Test to open dummy document working

    * Scenarios for bookmark and redaction

    * Added scenario that tests comments feature on MV

    * EUI 8472 buffer/codecept migration new (#3060)

    * Updated changes to the back end

    * Commmented out the old ways that the actions were being placed given the permissions

    * commit changes to the permissions of the mock data in feature file

    * formatting

    * Fix the feature file to have passing tests

    * Fix the feature file to have passing tests

    * Fix the feature file to have passing tests

    * EUI 8419 manage links testing (#3073)

    * commit changes to the permissions of the mock data in feature file

    * formatting

    * Fix the feature file to have passing tests

    * Fix the feature file to have passing tests

    * Fix the feature file to have passing tests

    * integration tests branch to buffer (#3063)

    * 4/6 global search tests passing

    * Added mock data for sherlock holmes

    * Search page field validation is now passing

    * Test

    * Pagination test passing

    * fixed search page and pagination tests

    * 6/8 passing

    * cleaning up tags

    * taskActions.feature all tests passing

    * EUI-8488 fix tests tab (#3081)

    * fix tests tab

    * Task -> 7 passing
    Find fix to "assigned to" on ui

    * Task -> 7 passing
    Find fix to "assigned to" on ui

    * in progress, users.js update to cw

    * task tab test updated

    * updated priority, tests passing

    * formatting

    * formatting

    ---------

    Co-authored-by: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>

    * E UI 8423 will changes (#3097)

    * EUI-8423 BOOKINGS

    * Integration tests booking UI mock setup fixed

    * Work access tests 6/7

    * Dates updated all tests passing

    ---------

    Co-authored-by: AbdullahShahHMCTS <Abdullah.shah@hmcts.net>
    Co-authored-by: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>

    * add suppressions

    * Pass buffer branch tests (#3104)

    * updated codecept -> integration testing

    * updated codecept -> functional testing

    * updated functional tests to passing

    * add @functional_enabled tag

    * add @functional_enabled tag

    * suppressions

    * typo not running e2e's - fixed

    * All work filters tests fix (#3107)

    * fix failing tests

    * suppressions

    ---------

    Co-authored-by: WillOrchard <127200688+WillOrchard@users.noreply.github.com>
    Co-authored-by: sreekanthpuligadda <sreekanth.puligadda@hmcts.net>

    ---------

    Co-authored-by: WillOrchard <127200688+WillOrchard@users.noreply.github.com>
    Co-authored-by: AbdullahShahHMCTS <77341369+AbdullahShahHMCTS@users.noreply.github.com>

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: test_codecept/backendMock/services/globalSearch/routes.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const service = require('./index')
+
+const userApiData = require('../userApiData')
+
+const jwt = require('jsonwebtoken');
+
+router.post('/', (req, res) => {
+
+    // res.status(200).send(service.getDefaulstResponse());
+    userApiData.sendResponse(req, res, "GlobalSearchResults", () => service.getDefaulstResponse())
```
Comment (Daniel-Lam) [inline test_codecept/backendMock/services/globalSearch/routes.js:15]: @sreekanthpuligadda - looks like a typo... should it be `getDefaultResponse()`?

File: test_codecept/backendMock/services/hearings/routes.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+const express = require('express')
+
+const router = express.Router({ mergeParams: true });
+const service = require('./index')
+
+const userApiData = require('../userApiData')
+
+
+router.get('/hearings/:caseId', (req, res) => {
+
```
Comment (Daniel-Lam) [inline test_codecept/backendMock/services/hearings/routes.js:12]: @sreekanthpuligadda - lot of whitespace here

File: test_codecept/backendMock/services/prd/index.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+const hearingTypes = [
+
+]
+
+class PRDApi{
+    
+
+    constructor(){
+
```
Comment (Daniel-Lam) [inline test_codecept/backendMock/services/prd/index.js:14]: @sreekanthpuligadda - lot of whitespace here

File: test_codecept/backendMock/services/prdOrganisations/routes.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (Daniel-Lam) [inline test_codecept/backendMock/services/prdOrganisations/routes.js:1]: @sreekanthpuligadda - whitespace (as before)

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@sreekanthpuligadda - I wasn't able to go through everything but in general, I noticed two things:

- Lots of whitespace in some files
- A fair amount of commented-out code

Also, does `yarn lint` cover the `test_codecept` directory? There is a bit of formatting inconsistency.

**Review by Daniel-Lam — APPROVED** [review]

**Review by DavidJayakumar — APPROVED** [review]

**Review by udaydenduluri33 — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by DavidJayakumar — APPROVED** [review]

**Pull request closed without merging**